discover/iterator.go

```diff
@@ -35,3 +37,56 @@ func generateIterators(list []string) []*Iterator {
 
 	return result
 }
+
+func findCommonDirectory(effxFileLocations []string) string {
+	matchedEffxFiles := generateIterators(effxFileLocations)
+	prefixString := ""
+
+	for len(matchedEffxFiles) > 0 {
+		count := make(map[string]int)
+		for _, matchedFile := range matchedEffxFiles {
+			peek := matchedFile.Peek()
+			if peek != "" {
+				count[peek]++
+			}
+		}
+
+		maxK := ""
+		maxV := 1
+		for k, v := range count {
+			if v > maxV {
+				maxK = k
+				maxV = v
+			}
+		}
+
+		nextRound := make([]*Iterator, 0, maxV)
+		for _, matchedFile := range matchedEffxFiles {
+			// advance ptr
+			if matchedFile.Next() == maxK {
```

Comment:
we should short circuit here with something like `&& maxCharCount > len(matchedEffxFiles) / 2`. This will prevent us from going too far down a given path... e.g. eventually, you have two services that start with the same character.